### PR TITLE
Add a weather forecast plugin

### PR DESCRIPTION
**Pull request by roygbyte**
This is a little plugin that pulls weather data from WeatherAPI. 

Might be more fitting for the contrib repo. Open to suggestions!

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8126)

### CODE REVIEW

**Comment by hius07** [comment]
CircleCI test:
Warning: tab character detected. Please use spaces.

File: plugins/weather.koplugin/_meta.lua [hunk 1]
```diff
@@ -0,0 +1,6 @@
+local _ = require("gettext")
+return {
+    name = "weather",
+    fullname = _("Weather"),
+    description = _([[Get the feather forcast]]),
```
Comment (Frenzie) [inline plugins/weather.koplugin/_meta.lua:5]: typo btw

**Comment by Frenzie** [comment]
I'm inclined to agree with contrib but I don't want to think about it too much right now. ^_^

**Comment by Galunid** [comment]
+1 for contrib

**Comment by poire-z** [comment]
Some screenshots at https://github.com/roygbyte/koweather , with the essential info that I would be fine with.

I think I had sometimes wished to have some weather info on my Kobo - but I don't really remember why/when, because we mostly all have a phone handy that shows more, with pictures and maps :)
My previous opinions on non-reading related plugins: https://github.com/koreader/koreader/pull/5854#issuecomment-586579954 https://github.com/koreader/koreader/pull/6848#issuecomment-722027294

So, yes, probably contrib - and also for a few more reasons:
- it depends on some external API and token: so, some manual work for the user
- external API means it's not standalone and will need more maintenance (when the API change), and may become easily non-functionnal - and need more maintenance/rework
- would it work with a French postal code ? Probably not quickly reading the API docs - but a city name work just fine - or would need to show a list to pick one from. Might need more UI work for easier setup.
- bit too plain text, I'd like to see some sun/moon/clouds icons ! which could be fun to add (even code-generated SVG from these numbers :)
- other people may wish to use a different provider/API, or show the info in various other ways - having this in the main repo would make it THE meteo plugin - having it in contrib would allow more implementations so one can pick the best for his needs.
- your coding style ( `if(` , 3-space indents, unnecessary parens) differs from koreader's core, less harassment about that from us in contrib :)

**Comment by roygbyte** [comment]
@poire-z, I totally see where you're coming from.

> My previous opinions on non-reading related plugins: [#5854 (comment)](https://github.com/koreader/koreader/pull/5854#issuecomment-586579954) [#6848 (comment)](https://github.com/koreader/koreader/pull/6848#issuecomment-722027294)

Having understood your thoughts, and like, the aim of the project, I totally agree it is better placed in the contrib repo. All of your other points are entirely valid, too. Regarding your comment about keeping KO fully baked as a hot and ready reading app (https://github.com/koreader/koreader/pull/5854#issuecomment-586579954), I have this silly vision of using my eReader as a substitute for the morning paper. And hence: having a weather plugin fits into that scope for my own personal reading journey.

>     * bit too plain text, I'd like to see some sun/moon/clouds icons ! which could be fun to add (even code-generated SVG from these numbers :)
> 
I would *love* to have it look more illustrated. This is my first project in Lua, so I'm still getting the hang of things. 

>     * your coding style ( `if(` , 3-space indents, unnecessary parens) differs from koreader's core, less harassment about that from us in contrib :)

I don't minding following the styles. I just hadn't figured out what they were yet :stuck_out_tongue:

**Comment by Frenzie** [comment]
The style is detailed here: https://github.com/koreader/koreader-base/wiki/Coding-style

**Comment by roygbyte** [comment]
Closing this because I'm going to move 'er over to contrib.

**Comment by poire-z** [comment]
Btw, given that you already have a proper project on your own user space, you could just do as @zwim did for his https://github.com/koreader/contrib/tree/main/calculator.koplugin : just add to koreader/contrib a directory + README linking to your own project.
That way, you're more free to update it and add commits as it's your own project - using koreader/contrib/ for the main code will require us to review and merge these commits into koreader/contrib.

**Comment by poire-z** [comment]
> I have this silly vision of using my eReader as a substitute for the morning paper. And hence: having a weather plugin fits into that scope for my own personal reading journey. Obviously, that journey is different from the KOReader project's. 

I'm not sure this could make it into core, but we could have a plugin that generates and EPUB (same filename) with bits from here and there (weather, horoscope, news...), your own newspaper :) That would make it a bit more "reading" oriented, generating a book (instead of having a weather plugin, a horoscope plugin, using our limited UI widgets).
(That's what our newsdownloader plugin does a bit, but this plugin is halk-baked/half-working, too dependant on the remote servers quirks...)

**Comment by roygbyte** [comment]
> weather, horoscope, news...), your own newspaper :) That would make it a bit more "reading" oriented, generating a book (instead of having a weather plugin, a horoscope plugin, using our limited UI widgets).

@poire-z Yes, that is exactly what I want to wake up to in the morning—weather, news, horoscopes, and maybe other content like yesterday's bookmarks and reading stats. I hadn't considered delivering that *as* an ebook (I mostly envisioned it as being implemented with the UI widgets.) But in some ways, that's actually genius. I will go through the newsdownloader plugin you mentioned and see where the effort has lead. 🙈

**Pull request closed without merging**